standard/lpdb/condition.lua

```diff
@@ -13,21 +13,28 @@ local Array = require('Module:Array')
 local Condition = {}
 
 -- Abstract class, node of the conditions tree
+---@class _ConditionNode
+---@field is_a function
 local _ConditionNode = Class.new()
 
+---@enum lpdbBooleanOperator
+---| '"AND"'
+---| '"OR"'
+
 ---A tree of conditions, specifying the conditions for an LPDB request.
 ---Can be used recursively, as in, a tree of trees.
----@class ConditionTree
+---@class ConditionTree:_ConditionNode
+---@operator call: fun(booleanOperator: lpdbBooleanOperator)
 ---@field _nodes ConditionNode[]
----@field booleanOperator 'AND'|'OR'
+---@field booleanOperator lpdbBooleanOperator
 local ConditionTree = Class.new(_ConditionNode,
 	function(self, booleanOperator)
 		self.booleanOperator = booleanOperator
 		self._nodes = {}
 	end
 )
 
----@param node ConditionNode|ConditionNode[]
+---@param node _ConditionNode|ConditionNode[]
```

Comment:
```suggestion
---@param node _ConditionNode[]
```
should be correct

Reply:
nope, it can also eat a node directly, doesn't need to be an array of nodes
it also throws a warning with this
but the following would be correct:
```suggestion
---@param node _ConditionNode|_ConditionNode[]
```